packages/cli/src/cli/target.rs

```diff
@@ -50,3 +50,127 @@ pub(crate) struct TargetArgs {
     #[clap(long)]
     pub(crate) target: Option<String>,
 }
+
+impl TargetArgs {
+    pub(crate) fn arch(&self) -> Arch {
+        self.arch.unwrap_or_default()
+    }
+}
+
+#[derive(Debug, Default, Copy, Clone, PartialEq, Deserialize, clap::ValueEnum)]
+#[non_exhaustive]
+pub(crate) enum Arch {
+    // Android: armv7l, armv7-linux-androideabi
+    Arm,
+    // Android: aarch64, aarch64-linux-android
+    #[default]
+    Arm64,
+    // Android: i386, i686-linux-android
+    X86,
+    // Android: x86_64, x86_64-linux-android
+    X64,
+}
+
+impl Arch {
+    pub(crate) fn autodetect() -> Option<Self> {
+        // Try auto detecting arch through adb.
+        static AUTO_ARCH: OnceCell<Option<Arch>> = OnceCell::new();
+
+        *AUTO_ARCH.get_or_init(|| {
+            // TODO: Wire this up with --device flag. (add `-s serial`` flag before `shell` arg)
+            let output = std::process::Command::new("adb")
+                .arg("shell")
+                .arg("uname")
+                .arg("-m")
+                .output();
```

Comment:
We can't use std::process since this blocks the thread - can we somehow reorient this to be async + lazy?